### PR TITLE
[refactor] test utils: simplify signatures and remove from ruff exclusion list.

### PR DESCRIPTION
**Pull request by jayaddison**
### Feature or Bugfix
- Refactoring

### Purpose
- Expand `ruff` formatting through the codebase.

### Detail
- Apply some small method-signature cleanups in `tests/utils.py`.
- Remove `tests/utils.py` from the `ruff` exclusion list.

### Relates
- Follow-up to #12146 to remove `tests/utils.py` from the `ruff` exclusion list.

### CODE REVIEW

**Review by danieleades — APPROVED** [review]
there are broader changes here than just formatting-  i can't comment on those, but the formatting looks good

**Comment by jayaddison** [comment]
Understood - thank you!

File: tests/utils.py [hunk 1]
```diff
@@ -14,22 +14,22 @@
...
+CERT_FILE: Final[str] = str(TESTS_ROOT / 'certs' / 'cert.pem')
 
 # File lock for tests
 LOCK_PATH: Final[str] = str(TESTS_ROOT / 'test-server.lock')
 
 
 class HttpServerThread(Thread):
-    def __init__(self, handler: type[BaseRequestHandler], /, *args: Any, **kwargs: Any) -> None:
-        super().__init__(*args, **kwargs)
-        self.server = ThreadingHTTPServer(("localhost", 7777), handler)
```
Comment (chrisjsewell) [inline tests/utils.py:32]: I guessing this is not actually part of ruff's formatting, to remove `args`, `kwargs`?

Comment (jayaddison) [reply tests/utils.py:32]: Correct, yep; we don't use them, and it made the eventual formatted output more concise to remove them.

**Comment by chrisjsewell** [comment]
> there are broader changes here than just formatting

yehh, it seems overkill to insist these are two separate PRs, but maybe at least just change the PR title, to reflect that actually most of the changes are not to do with formatiing 😅

**Review by chrisjsewell — APPROVED** [review]
cheers

**Comment by jayaddison** [comment]
Thanks!

**Pull request merged**